src/components/pages/maintenance/ruleDetails/ruleDetails.js

```diff
@@ -65,19 +68,23 @@ export class RuleDetails extends Component {
     this.subscriptions.push(
       this.restartTelemetry$
         .distinctUntilChanged()
-        .filter(deviceIds => deviceIds)
         .map(deviceIds => deviceIds.split(idDelimiter))
         .do(() => this.setState({ telemetry: {}, telemetryIsPending: false }))
-        .switchMap(deviceIds =>
-          TelemetryService.getTelemetryByDeviceIdP15M(deviceIds)
-            .merge(
-              this.telemetryRefresh$ // Previous request complete
-                .delay(Config.telemetryRefreshInterval) // Wait to refresh
-                .do(onPendingStart)
-                .flatMap(_ => TelemetryService.getTelemetryByDeviceIdP1M(deviceIds))
-            )
-            .flatMap(transformTelemetryResponse(() => this.state.telemetry))
-            .map(telemetry => ({ telemetry, telemetryIsPending: false }))
+        .switchMap(deviceIds => {
+          if (deviceIds[0]) {
```

Comment:
Why check the initial device id instead of checking the length?

Reply:
The actual value is [""].  So... The array length is 1.  The length of the first item in the array is 0.

Reply:
I went with the most compact javascript syntax that got the job done... which makes it about as readable as other compact javascript syntax.  Hahaha.

if (deviceIds[0]) {

is much more compact than

if (deviceIds.length === 1 && deviceIds[0].length === 0) {